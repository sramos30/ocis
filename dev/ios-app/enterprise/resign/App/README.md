Place your `unsigned.ipa` file in this folder, which should be signed.

After the resigning process, here you will find the `signed.ipa` file.

