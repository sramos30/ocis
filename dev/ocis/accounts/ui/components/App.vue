<template>
  <div>
    <main class="oc-flex oc-flex-column oc-height-1-1 oc-p-m" id="accounts-app">
      <template v-if="isInitialized">
        <h1 class="oc-invisible-sr">
          <translate>Accounts</translate>
        </h1>
        <div class="oc-app-bar">
          <accounts-batch-actions
            v-if="isAnyAccountSelected"
            :number-of-selected-accounts="numberOfSelectedAccounts"
            :selected-accounts="selectedAccounts"
          />
          <accounts-create v-else />
        </div>
        <oc-grid class="oc-flex-1 oc-overflow-auto">
          <div class="oc-width-expand">
            <accounts-list :accounts="accounts" />
          </div>
        </oc-grid>
      </template>
      <template v-else-if="hasFailed">
        <oc-alert
          variation="warning"
          no-close
          class="oc-m"
          id="accounts-list-loading-failed"
        >
          <oc-icon
            name="error-warning"
            variation="warning"
            class="oc-float-left oc-mr-s"
          />
          <translate>You don't have permissions to manage accounts.</translate>
        </oc-alert>
      </template>
      <oc-loader id="accounts-list-loader" v-else />
    </main>
  </div>
</template>

<script>
import { mapGetters, mapActions, mapState } from 'vuex'
import AccountsList from './accounts/AccountsList.vue'
import AccountsCreate from './accounts/AccountsCreate.vue'
import AccountsBatchActions from './accounts/AccountsBatchActions.vue'

export default {
  name: 'App',
  components: { AccountsBatchActions, AccountsList, AccountsCreate },
  computed: {
    ...mapGetters('Accounts', [
      'isInitialized',
      'hasFailed',
      'getAccountsSorted',
      'isAnyAccountSelected'
    ]),
    ...mapState('Accounts', ['selectedAccounts']),

    accounts () {
      return this.getAccountsSorted
    },
    numberOfSelectedAccounts () {
      return this.selectedAccounts.length
    }
  },
  methods: {
    ...mapActions('Accounts', ['initialize'])
  },
  created () {
    this.initialize()
  }
}
</script>
