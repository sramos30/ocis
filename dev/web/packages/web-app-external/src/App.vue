<template>
  <main
    class="oc-height-1-1"
    :class="{
      'oc-flex oc-flex-center oc-flex-middle': loading || loadingError
    }"
  >
    <h1 class="oc-invisible-sr" v-text="pageTitle" />
    <loading-screen v-if="loading" />
    <error-screen v-else-if="loadingError" :message="errorMessage" />
    <iframe
      v-if="appUrl && method === 'GET'"
      :src="appUrl"
      class="oc-width-1-1 oc-height-1-1"
      :title="iFrameTitle"
      allowfullscreen
    />
    <div v-if="appUrl && method === 'POST' && formParameters" class="oc-height-1-1">
      <form :action="appUrl" target="app-iframe" method="post">
        <input ref="subm" type="submit" :value="formParameters" class="oc-hidden" />
        <div v-for="(item, key, index) in formParameters" :key="index">
          <input :name="key" :value="item" type="hidden" />
        </div>
      </form>
      <iframe
        name="app-iframe"
        class="oc-width-1-1 oc-height-1-1"
        :title="iFrameTitle"
        allowfullscreen
      />
    </div>
  </main>
</template>

<script>
import { mapGetters } from 'vuex'
import ErrorScreen from './components/ErrorScreen.vue'
import LoadingScreen from './components/LoadingScreen.vue'

// FIXME: hacky, get rid asap, just a workaround
// same as packages/web-app-files/src/views/PublicFiles.vue
const unauthenticatedUserReady = async (router, store) => {
  if (store.getters.userReady) {
    return
  }

  const publicToken = router.currentRoute.query['public-token']
  const publicLinkPassword = store.getters['Files/publicLinkPassword']

  await store.dispatch('loadCapabilities', {
    publicToken,
    ...(publicLinkPassword && { user: 'public', password: publicLinkPassword })
  })

  store.commit('SET_USER_READY', true)
}

export default {
  name: 'ExternalApp',

  components: {
    ErrorScreen,
    LoadingScreen
  },

  data: () => ({
    loading: false,
    loadingError: false,
    errorMessage: '',
    appUrl: '',
    method: '',
    formParameters: {}
  }),
  computed: {
    ...mapGetters(['getToken', 'capabilities', 'configuration']),
    ...mapGetters('Files', ['publicLinkPassword']),

    pageTitle() {
      const translated = this.$gettext('"%{appName}" app page')
      return this.$gettextInterpolate(translated, {
        appName: this.appName
      })
    },
    iFrameTitle() {
      const translated = this.$gettext('"%{appName}" app content area')
      return this.$gettextInterpolate(translated, {
        appName: this.appName
      })
    },
    appName() {
      return this.$route.params.appName
    },
    fileId() {
      return this.$route.params.fileId
    }
  },
  async created() {
    await unauthenticatedUserReady(this.$router, this.$store)

    this.loading = true

    // build headers with respect to the actual auth situation
    const { 'public-token': publicToken } = this.$route.query
    const publicLinkPassword = this.publicLinkPassword
    const accessToken = this.getToken
    const headers = {
      'X-Requested-With': 'XMLHttpRequest',
      ...(publicToken && { 'public-token': publicToken }),
      ...(publicLinkPassword && {
        Authorization:
          'Basic ' + Buffer.from(['public', publicLinkPassword].join(':')).toString('base64')
      }),
      ...(accessToken && {
        Authorization: 'Bearer ' + accessToken
      })
    }

    // fetch iframe params for app and file
    const configUrl = this.configuration.server
    const appOpenUrl = this.capabilities.files.app_providers[0].open_url.replace(/^\/+/, '')
    const url =
      configUrl +
      appOpenUrl +
      `?file_id=${this.fileId}` +
      (this.appName ? `&app_name=${this.appName}` : '')

    const response = await fetch(url, {
      method: 'POST',
      headers
    })

    if (response.status !== 200) {
      this.errorMessage = response.message
      this.loading = false
      this.loadingError = true
      console.error('Error fetching app information', response.status, this.errorMessage)
      return
    }

    const data = await response.json()

    if (!data.app_url || !data.method) {
      this.errorMessage = this.$gettext('Error in app server response')
      this.loading = false
      this.loadingError = true
      console.error('Error in app server response')
      return
    }

    this.appUrl = data.app_url
    this.method = data.method
    if (data.form_parameters) this.formParameters = data.form_parameters

    if (this.method === 'POST' && this.formParameters) {
      this.$nextTick(() => this.$refs.subm.click())
    }
    this.loading = false
  }
}
</script>
