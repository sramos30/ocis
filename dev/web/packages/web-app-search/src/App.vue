<template>
  <main id="search" class="oc-flex oc-height-1-1">
    <router-view id="search-view" />
  </main>
</template>
<script>
export default {
  name: 'Search'
}
</script>
