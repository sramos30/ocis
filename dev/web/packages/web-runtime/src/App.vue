<template>
  <div id="web">
    <oc-hidden-announcer :announcement="announcement" level="polite" />
    <skip-to target="web-content-main">
      <translate>Skip to main</translate>
    </skip-to>
    <component :is="layout"></component>
    <portal-target name="app.runtime.modal" multiple></portal-target>
    <oc-modal
      v-if="modal.displayed"
      :variation="modal.variation"
      :icon="modal.icon"
      :title="modal.title"
      :message="modal.message"
      :has-input="modal.hasInput"
      :input-label="modal.inputLabel"
      :input-disabled="modal.inputDisabled"
      :input-value="modal.inputValue"
      :input-description="modal.inputDescription"
      :input-error="modal.inputError"
      :button-cancel-text="modal.cancelText"
      :button-confirm-text="modal.confirmText"
      :button-confirm-disabled="modal.confirmDisabled || !!modal.inputError"
      @cancel="modal.onCancel"
      @confirm="modal.onConfirm"
      @input="modal.onInput"
      @mounted="focusModal"
      @beforeDestroy="focusModal"
    />
  </div>
</template>
<script lang="ts">
import { mapGetters, mapState, mapActions } from 'vuex'
import SkipTo from './components/SkipTo.vue'
import LayoutApplication from './layouts/Application.vue'
import LayoutLoading from './layouts/Loading.vue'
import LayoutPlain from './layouts/Plain.vue'
import { getBackendVersion, getWebVersion } from './container/versions'
import { defineComponent } from '@vue/composition-api'

export default defineComponent({
  components: {
    SkipTo
  },
  data() {
    return {
      $_notificationsInterval: null,
      windowWidth: 0,
      announcement: '',
      activeBlobStyle: {}
    }
  },
  computed: {
    ...mapState(['route', 'user', 'modal', 'sidebar']),
    ...mapGetters(['configuration', 'capabilities', 'getSettingsValue']),
    layout() {
      if (this.user.isAuthenticated && !this.user.userReady) {
        return LayoutLoading
      }

      if (
        !this.$route.name ||
        [
          'login',
          'oidcCallback',
          'oidcSilentRedirect',
          'privateLink',
          'publicLink',
          'accessDenied'
        ].includes(this.$route.name)
      ) {
        return LayoutPlain
      }

      return LayoutApplication
    },
    favicon() {
      return this.configuration.currentTheme.logo.favicon
    },

    selectedLanguage() {
      return this.getSettingsValue({
        extension: 'ocis-accounts',
        bundle: 'profile',
        setting: 'language'
      })
    }
  },
  watch: {
    $route: {
      immediate: true,
      handler: function (to) {
        this.announceRouteChange(to)
        document.title = this.extractPageTitleFromRoute(to)
      }
    },
    capabilities: {
      immediate: true,
      handler: function (caps) {
        if (!caps?.notifications) {
          return
        }

        // setup periodic loading of notifications if the server supports them
        this.$nextTick(() => {
          this.$_updateNotifications()
        })
        this.$_notificationsInterval = setInterval(() => {
          this.$_updateNotifications()
        }, 30000)
      }
    },
    selectedLanguage: {
      immediate: true,
      handler(language) {
        let languageCode = this.$language.defaultLanguage
        if (language !== null && language.listValue.values.length > 0) {
          languageCode = language.listValue.values[0].stringValue
        }
        if (languageCode) {
          this.$language.current = languageCode
          document.documentElement.lang = languageCode
        }
      }
    }
  },

  destroyed() {
    if (this.$_notificationsInterval) {
      clearInterval(this.$_notificationsInterval)
    }
  },

  metaInfo() {
    const metaInfo: any = {}
    if (this.favicon) {
      metaInfo.link = [{ rel: 'icon', href: this.favicon }]
    }
    const metaGenerator = {
      name: 'generator',
      content: [getWebVersion(), getBackendVersion({ store: this.$store })]
        .filter(Boolean)
        .join(', ')
    }
    metaInfo.meta = [metaGenerator]
    return metaInfo
  },

  methods: {
    ...mapActions(['fetchNotifications']),

    focusModal(component, event) {
      this.focus({
        revert: event === 'beforeDestroy'
      })
    },

    $_updateNotifications() {
      this.fetchNotifications(this.$client).catch((error) => {
        console.error('Error while loading notifications: ', error)
        clearInterval(this.$_notificationsInterval)
      })
    },

    announceRouteChange(route) {
      const pageTitle = this.extractPageTitleFromRoute(route, false)
      const translated = this.$gettext('Navigated to %{ pageTitle }')
      this.announcement = this.$gettextInterpolate(translated, { pageTitle })
    },

    extractPageTitleFromRoute(route, includeGeneralName = true) {
      const routeTitle = route.meta.title ? this.$gettext(route.meta.title) : route.name
      const titleSegments = [routeTitle]

      if (includeGeneralName) {
        titleSegments.push(this.configuration.currentTheme.general.name)
      }

      if (route.params.item) {
        if (route.name.startsWith('files-')) {
          const fileTree = route.params.item.split('/').filter((el) => el.length)

          if (fileTree.length) {
            titleSegments.unshift(fileTree.pop())
          }
        } else {
          titleSegments.unshift(route.params.item)
        }
      }

      return titleSegments.join(' - ')
    }
  }
})
</script>
<style lang="scss">
body {
  margin: 0;
}

#web {
  background-color: var(--oc-color-swatch-brand-default);
  height: 100vh;
  max-height: 100vh;
  overflow-y: hidden;
}

iframe {
  border: 0;
}
</style>
