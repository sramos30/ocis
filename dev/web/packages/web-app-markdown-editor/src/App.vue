<template>
  <main id="markdown-editor" class="oc-mx-s oc-my-s">
    <markdown-editor-app-bar
      :current-file-context="currentFileContext"
      :is-loading="isLoading"
      :is-dirty="isDirty"
      @closeApp="closeApp"
      @save="save"
    />
    <oc-notifications>
      <oc-notification-message
        v-if="lastError"
        :message="lastError"
        status="danger"
        @close="clearLastError"
      />
    </oc-notifications>
    <div class="oc-flex">
      <div class="oc-container oc-width-1-2">
        <oc-textarea
          id="markdown-editor-input"
          v-model="currentContent"
          name="input"
          full-width
          :label="$gettext('Editor')"
          class="oc-height-1-1"
          :rows="20"
        />
      </div>
      <div class="oc-container oc-width-1-2">
        <!-- eslint-disable-next-line vue/no-v-html -->
        <div id="markdown-editor-preview" v-html="renderedMarkdown" />
      </div>
    </div>
  </main>
</template>
<script>
import MarkdownEditorAppBar from './MarkdownEditorAppBar.vue'
import { useAppDefaults } from 'web-pkg/src/composables'
import { marked } from 'marked'
import sanitizeHtml from 'sanitize-html'
import { useTask } from 'vue-concurrency'
import { computed, getCurrentInstance, onMounted, ref, unref } from '@vue/composition-api'

export default {
  name: 'MarkdownEditor',
  components: {
    MarkdownEditorAppBar
  },
  setup() {
    const serverContent = ref()
    const currentContent = ref()
    const currentETag = ref()

    const loadFileTask = useTask(function* (signal, that) {
      const filePath = unref(that.currentFileContext).path
      return yield that.getFileContents(filePath).then((response) => {
        serverContent.value = currentContent.value = response.body
        currentETag.value = response.headers.ETag
        return response
      })
    }).restartable()

    const saveFileTask = useTask(function* (signal, that) {
      const filePath = unref(that.currentFileContext).path
      const newContent = unref(currentContent)

      return yield that
        .putFileContents(filePath, newContent, {
          previousEntityTag: unref(currentETag)
        })
        .then(
          (response) => {
            serverContent.value = newContent
            // FIXME: above we need response.headers.ETag, here we need response ETag - feels inconsistent
            currentETag.value = response.ETag
          },
          (error) => {
            lastError.value = error.message
          }
        )
    }).restartable()

    const lastError = ref()
    const clearLastError = () => {
      lastError.value = null
    }

    const renderedMarkdown = computed(() => {
      return unref(currentContent) ? sanitizeHtml(marked(unref(currentContent))) : null
    })

    const isDirty = computed(() => {
      return unref(serverContent) !== unref(currentContent)
    })

    const isLoading = computed(() => {
      return loadFileTask.isRunning || saveFileTask.isRunning
    })

    onMounted(() => {
      loadFileTask.perform(getCurrentInstance().proxy)
    })

    const save = function () {
      saveFileTask.perform(this)
    }

    return {
      ...useAppDefaults({
        applicationName: 'markdown-editor'
      }),

      // tasks
      loadFileTask,
      saveFileTask,

      // data
      isLoading,
      isDirty,
      currentContent,
      lastError,
      renderedMarkdown,

      // methods
      clearLastError,
      save
    }
  }
}
</script>
<style lang="scss" scoped>
#markdown-editor-preview {
  max-height: 80vh;
  overflow-y: scroll;
}
</style>
