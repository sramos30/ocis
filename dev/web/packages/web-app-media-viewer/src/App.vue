<template>
  <main
    id="mediaviewer"
    ref="mediaviewer"
    tabindex="-1"
    @keydown.left="prev"
    @keydown.right="next"
    @keydown.esc="closeApp"
  >
    <h1 class="oc-invisible-sr" v-text="pageTitle" />
    <div
      v-show="!loading && activeMediaFileCached"
      class="
        oc-width-1-1 oc-flex oc-flex-center oc-flex-middle oc-p-s oc-box-shadow-medium
        media-viewer-player
      "
    >
      <video v-if="medium.isVideo" :key="`media-video-${medium.id}`" controls preload>
        <source :src="medium.url" :type="`video/${medium.ext}`" />
      </video>
      <img
        v-else
        :key="`media-image-${medium.id}`"
        :src="medium.url"
        :alt="medium.name"
        :data-id="medium.id"
      />
    </div>
    <div v-if="loading" class="oc-position-center">
      <oc-spinner :aria-label="$gettext('Loading media file')" size="xlarge" />
    </div>
    <oc-icon
      v-if="failed"
      name="review"
      variation="danger"
      size="xlarge"
      class="oc-position-center"
      :accessible-label="$gettext('Failed to load media file')"
    />

    <div class="oc-position-medium oc-position-bottom-center media-viewer-details">
      <p
        class="oc-text-lead oc-text-center oc-text-truncate oc-p-s media-viewer-file-name"
        aria-hidden="true"
      >
        {{ medium.name }}
      </p>
      <div
        class="
          oc-background-brand
          oc-p-s
          oc-width-large
          oc-flex
          oc-flex-middle
          oc-flex-center
          oc-flex-around
          media-viewer-controls-action-bar
        "
      >
        <oc-button
          class="media-viewer-controls-previous"
          appearance="raw"
          variation="inverse"
          :aria-label="$gettext('Show previous media file in folder')"
          @click="prev"
        >
          <oc-icon size="large" name="arrow-drop-left" />
        </oc-button>
        <p v-if="!isFolderLoading" class="oc-m-rm media-viewer-controls-action-count">
          <span aria-hidden="true" v-text="ariaHiddenFileCount" />
          <span class="oc-invisible-sr" v-text="screenreaderFileCount" />
        </p>
        <oc-button
          class="media-viewer-controls-next"
          appearance="raw"
          variation="inverse"
          :aria-label="$gettext('Show next media file in folder')"
          @click="next"
        >
          <oc-icon size="large" name="arrow-drop-right" />
        </oc-button>
        <oc-button
          class="media-viewer-controls-download"
          appearance="raw"
          variation="inverse"
          :aria-label="$gettext('Download currently viewed file')"
          @click="downloadMedium"
        >
          <oc-icon size="large" name="file-download" fill-type="line" />
        </oc-button>
        <oc-button
          class="media-viewer-controls-close"
          appearance="raw"
          variation="inverse"
          :aria-label="$gettext('Close mediaviewer app')"
          @click="closeApp"
        >
          <oc-icon size="large" name="close" />
        </oc-button>
      </div>
    </div>
  </main>
</template>
<script>
import { mapGetters } from 'vuex'
import { useAppDefaults } from 'web-pkg/src/composables'

export default {
  name: 'Mediaviewer',
  setup() {
    return {
      ...useAppDefaults({
        applicationName: 'media'
      })
    }
  },
  data() {
    return {
      loading: true,
      failed: false,

      activeIndex: null,
      direction: 'rtl',

      medium: {},
      media: []
    }
  },

  computed: {
    ...mapGetters(['getToken', 'capabilities']),

    pageTitle() {
      const translated = this.$gettext('Mediaviewer for %{currentMediumName}')
      return this.$gettextInterpolate(translated, {
        currentMediumName: this.medium.name
      })
    },
    ariaHiddenFileCount() {
      const translated = this.$gettext('%{ displayIndex } of %{ availableMediaFiles }')
      return this.$gettextInterpolate(translated, {
        displayIndex: this.activeIndex + 1,
        availableMediaFiles: this.mediaFiles.length
      })
    },
    screenreaderFileCount() {
      const translated = this.$gettext('Media file %{ displayIndex } of %{ availableMediaFiles }')
      return this.$gettextInterpolate(translated, {
        displayIndex: this.activeIndex + 1,
        availableMediaFiles: this.mediaFiles.length
      })
    },
    mediaFiles() {
      if (!this.activeFiles) {
        return []
      }

      return this.activeFiles.filter((file) => {
        return file.extension.toLowerCase().match(/(png|jpg|jpeg|gif|mp4|webm|ogg)/)
      })
    },
    activeMediaFile() {
      return this.mediaFiles[this.activeIndex]
    },
    activeMediaFileCached() {
      const cached = this.media.find((i) => i.id === this.activeMediaFile.id)
      return cached !== undefined ? cached : false
    },
    thumbDimensions() {
      switch (true) {
        case window.innerWidth <= 1024:
          return 1024
        case window.innerWidth <= 1280:
          return 1280
        case window.innerWidth <= 1920:
          return 1920
        case window.innerWidth <= 2160:
          return 2160
        default:
          return 3840
      }
    },
    thumbUrl() {
      const query = {
        x: this.thumbDimensions,
        y: this.thumbDimensions,
        // strip double quotes from etag
        // we have no etag, e.g. on shared with others page
        c: this.activeMediaFile.etag?.substr(1, this.activeMediaFile.etag.length - 2),
        scalingup: 0,
        preview: 1,
        a: 1
      }

      return this.getUrlForResource(this.activeMediaFile, query)
    },
    rawMediaUrl() {
      return this.getUrlForResource(this.activeMediaFile)
    },

    videoExtensions() {
      return ['mp4', 'webm', 'ogg']
    },

    isActiveMediaFileTypeVideo() {
      return this.videoExtensions.includes(this.activeMediaFile.extension.toLowerCase())
    },

    isActiveMediaFileTypeImage() {
      return !this.isActiveMediaFileTypeVideo
    },

    isUrlSigningEnabled() {
      return this.capabilities.core && this.capabilities.core['support-url-signing']
    }
  },

  watch: {
    activeIndex(o, n) {
      if (o !== n) {
        this.loadMedium()
      }
    }
  },

  async mounted() {
    // keep a local history for this component
    window.addEventListener('popstate', this.handleLocalHistoryEvent)
    await this.loadFolderForFileContext(this.currentFileContext)
    this.setCurrentFile(this.currentFileContext.path)
    this.$refs.mediaviewer.focus()
  },

  beforeDestroy() {
    window.removeEventListener('popstate', this.handleLocalHistoryEvent)

    this.media.forEach((medium) => {
      window.URL.revokeObjectURL(medium.url)
    })
  },

  methods: {
    setCurrentFile(filePath) {
      for (let i = 0; i < this.mediaFiles.length; i++) {
        if (this.mediaFiles[i].webDavPath === filePath) {
          this.activeIndex = i
          break
        }
      }
    },

    // react to PopStateEvent ()
    handleLocalHistoryEvent() {
      const result = this.$router.resolve(document.location)
      this.setCurrentFile(result.route.params.filePath)
    },

    // update route and url
    updateLocalHistory() {
      this.$route.params.filePath = this.activeMediaFile.webDavPath
      history.pushState({}, document.title, this.$router.resolve(this.$route).href)
    },

    loadMedium() {
      this.loading = true

      // Don't bother loading if files are cached
      if (this.activeMediaFileCached) {
        setTimeout(
          () => {
            this.medium = this.activeMediaFileCached
            this.loading = false
          },
          // Delay to animate
          50
        )
        return
      }

      // Fetch media
      const url = this.isActiveMediaFileTypeImage ? this.thumbUrl : this.rawMediaUrl
      // FIXME: at the moment the signing key is not cached, thus it will be loaded again on each request.
      // workaround for now: Load file as blob for images, load as signed url (if supported) for everything else.
      let promise
      if (this.isActiveMediaFileTypeImage || !this.isUrlSigningEnabled || !this.$route.meta.auth) {
        promise = this.mediaSource(url, 'url', null)
      } else {
        promise = this.$client.signUrl(url, 86400) // Timeout of the signed URL = 24 hours
      }

      promise
        .then((mediaUrl) => {
          this.media.push({
            id: this.activeMediaFile.id,
            name: this.activeMediaFile.name,
            url: mediaUrl,
            ext: this.activeMediaFile.extension,
            isVideo: this.isActiveMediaFileTypeVideo,
            isImage: this.isActiveMediaFileTypeImage
          })
          this.medium = this.activeMediaFileCached
          this.loading = false
          this.failed = false
        })
        .catch((e) => {
          this.loading = false
          this.failed = true
          console.error(e)
        })
    },

    downloadMedium() {
      if (this.loading) {
        return
      }

      return this.downloadFile(this.mediaFiles[this.activeIndex], this.isPublicLinkContext)
    },
    next() {
      if (this.loading) {
        return
      }
      this.failed = false
      this.direction = 'rtl'
      if (this.activeIndex + 1 >= this.mediaFiles.length) {
        this.activeIndex = 0
        return
      }
      this.activeIndex++
      this.updateLocalHistory()
    },
    prev() {
      if (this.loading) {
        return
      }
      this.failed = false
      this.direction = 'ltr'
      if (this.activeIndex === 0) {
        this.activeIndex = this.mediaFiles.length - 1
        return
      }
      this.activeIndex--
      this.updateLocalHistory()
    }
  }
}
</script>

<style lang="scss" scoped>
.media-viewer-player {
  max-width: 90vw;
  height: 70vh;
  margin: 10px auto;
  object-fit: contain;
  img,
  video {
    max-width: 85vw;
    max-height: 65vh;
  }
}

.media-viewer-controls-action-count {
  color: var(--oc-color-swatch-inverse-default);
}

@media (max-width: 959px) {
  .media-viewer-player {
    max-width: 100vw;
  }

  .media-viewer-details {
    left: 0;
    margin: 0;
    max-width: 100%;
    transform: none !important;
    width: 100%;
    .media-viewer-controls-action-bar {
      width: 100%;
    }
  }
}
</style>
