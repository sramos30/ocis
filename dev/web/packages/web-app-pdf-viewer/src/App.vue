<template>
  <main>
    <loading-screen v-if="loading" />
    <error-screen v-else-if="loadingError" />
    <object
      v-else
      class="pdf-viewer oc-width-1-1 oc-height-1-1"
      :data="blobUrl"
      type="application/pdf"
    />
  </main>
</template>
<script>
import { mapGetters } from 'vuex'
import { useAppDefaults } from 'web-pkg/src/composables'
import ErrorScreen from './components/ErrorScreen.vue'
import LoadingScreen from './components/LoadingScreen.vue'

export default {
  name: 'PDFViewer',
  components: {
    ErrorScreen,
    LoadingScreen
  },
  setup() {
    return {
      ...useAppDefaults({
        applicationName: 'pdf-viewer'
      })
    }
  },
  data: () => ({
    loading: true,
    loadingError: false,
    filePath: '',
    blobUrl: ''
  }),
  computed: {
    ...mapGetters(['getToken'])
  },
  created() {
    this.loadPdf(this.currentFileContext)
  },
  unmounted() {
    this.unloadPdf()
  },
  methods: {
    async loadPdf(fileContext) {
      try {
        this.loading = true
        const response = await this.getFileContents(fileContext.path, { responseType: 'blob' })
        this.blobUrl = URL.createObjectURL(response.body)
      } catch (e) {
        this.loadingError = true
        console.error('Error fetching pdf', e)
      } finally {
        this.loading = false
      }
    },
    unloadPdf() {
      URL.revokeObjectURL(this.blobUrl)
    }
  }
}
</script>
<style scoped>
.pdf-viewer {
  border: none;
  margin: 0;
  padding: 0;
  overflow: hidden;
}
</style>
