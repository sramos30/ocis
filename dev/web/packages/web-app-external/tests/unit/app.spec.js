import App from '../../src/App.vue'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import GetTextPlugin from 'vue-gettext'

const localVue = createLocalVue()

localVue.use(GetTextPlugin, {
  translations: 'does-not-matter.json',
  silent: true
})
localVue.use(Vuex)

const componentStubs = {
  ErrorScreen: true,
  LoadingScreen: true
}

const $route = {
  query: {
    'public-token': 'a-token'
  },
  params: {
    appName: 'exampleApp',
    fileId: '2147491323'
  }
}

const storeOptions = {
  getters: {
    getToken: jest.fn(() => 'GFwHKXdsMgoFwt'),
    configuration: jest.fn(() => ({
      server: 'http://example.com/'
    })),
    userReady: () => true,
    capabilities: jest.fn(() => ({
      files: {
        app_providers: [
          {
            apps_url: '/app/list',
            enabled: true,
            open_url: '/app/open'
          }
        ]
      }
    }))
  },
  modules: {
    External: {
      namespaced: true,
      getters: {
        mimeTypes: jest.fn()
      },
      actions: {
        fetchMimeTypes: jest.fn()
      },
      mutations: {
        SET_MIME_TYPES: jest.fn()
      }
    }
  }
}

const appUrl = 'https://example.test/d12ab86/loe009157-MzBw'

const providerSuccessResponsePost = {
  app_url: appUrl,
  method: 'POST',
  form_parameters: {
    access_token: 'asdfsadfsadf',
    access_token_ttl: '123456'
  }
}

const providerSuccessResponseGet = {
  app_url: appUrl,
  method: 'GET'
}

describe('The app provider extension', () => {
  beforeEach(() => {
    jest.spyOn(console, 'error').mockImplementation(() => {})
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should show a loading spinner while loading', async () => {
    global.fetch = jest.fn(() =>
      setTimeout(() => {
        Promise.resolve({
          ok: true,
          status: 200
        })
      }, 500)
    )
    const wrapper = createShallowMountWrapper()
    await wrapper.vm.$nextTick()
    expect(wrapper).toMatchSnapshot()
  })
  it('should show a meaningful message if an error occurs during loading', async () => {
    global.fetch = jest.fn(() =>
      Promise.resolve({
        ok: false,
        status: 500,
        message: 'We encountered an internal error'
      })
    )
    const wrapper = createShallowMountWrapper()
    await wrapper.vm.$nextTick()
    await wrapper.vm.$nextTick()
    expect(wrapper).toMatchSnapshot()
  })
  it('should fail for unauthenticated users', async () => {
    global.fetch = jest.fn(() =>
      Promise.resolve({
        ok: true,
        status: 401,
        message: 'Login Required'
      })
    )
    const wrapper = createShallowMountWrapper()
    await wrapper.vm.$nextTick()
    await wrapper.vm.$nextTick()
    expect(wrapper).toMatchSnapshot()
  })
  it('should be able to load an iFrame via get', async () => {
    global.fetch = jest.fn(() =>
      Promise.resolve({
        ok: true,
        status: 200,
        json: () => providerSuccessResponseGet
      })
    )

    const wrapper = createShallowMountWrapper()
    await wrapper.vm.$nextTick()
    await wrapper.vm.$nextTick()
    expect(wrapper).toMatchSnapshot()
  })
  it('should be able to load an iFrame via post', async () => {
    global.fetch = jest.fn(() =>
      Promise.resolve({
        ok: true,
        status: 200,
        json: () => providerSuccessResponsePost
      })
    )
    const wrapper = createShallowMountWrapper()
    await wrapper.vm.$nextTick()
    await wrapper.vm.$nextTick()
    expect(wrapper).toMatchSnapshot()
  })
})

function createShallowMountWrapper(options = {}) {
  return shallowMount(App, {
    localVue,
    store: createStore(),
    stubs: componentStubs,
    mocks: {
      $route
    },
    ...options
  })
}

function createStore() {
  return new Vuex.Store(storeOptions)
}
