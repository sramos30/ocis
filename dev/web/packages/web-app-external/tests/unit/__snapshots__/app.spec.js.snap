// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The app provider extension should be able to load an iFrame via get 1`] = `
<main class="oc-height-1-1">
  <h1 class="oc-invisible-sr">"exampleApp" app page</h1>
  <!----> <iframe src="https://example.test/d12ab86/loe009157-MzBw" title="&quot;exampleApp&quot; app content area" allowfullscreen="allowfullscreen" class="oc-width-1-1 oc-height-1-1"></iframe>
  <!---->
</main>
`;

exports[`The app provider extension should be able to load an iFrame via post 1`] = `
<main class="oc-height-1-1">
  <h1 class="oc-invisible-sr">"exampleApp" app page</h1>
  <!---->
  <!---->
  <div class="oc-height-1-1">
    <form action="https://example.test/d12ab86/loe009157-MzBw" target="app-iframe" method="post"><input type="submit" class="oc-hidden" value="[object Object]">
      <div><input name="access_token" type="hidden" value="asdfsadfsadf"></div>
      <div><input name="access_token_ttl" type="hidden" value="123456"></div>
    </form> <iframe name="app-iframe" title="&quot;exampleApp&quot; app content area" allowfullscreen="allowfullscreen" class="oc-width-1-1 oc-height-1-1"></iframe>
  </div>
</main>
`;

exports[`The app provider extension should fail for unauthenticated users 1`] = `
<main class="oc-height-1-1 oc-flex oc-flex-center oc-flex-middle">
  <h1 class="oc-invisible-sr">"exampleApp" app page</h1>
  <errorscreen-stub message="Login Required"></errorscreen-stub>
  <!---->
  <!---->
</main>
`;

exports[`The app provider extension should show a loading spinner while loading 1`] = `
<main class="oc-height-1-1 oc-flex oc-flex-center oc-flex-middle">
  <h1 class="oc-invisible-sr">"exampleApp" app page</h1>
  <loadingscreen-stub></loadingscreen-stub>
  <!---->
  <!---->
</main>
`;

exports[`The app provider extension should show a meaningful message if an error occurs during loading 1`] = `
<main class="oc-height-1-1 oc-flex oc-flex-center oc-flex-middle">
  <h1 class="oc-invisible-sr">"exampleApp" app page</h1>
  <errorscreen-stub message="We encountered an internal error"></errorscreen-stub>
  <!---->
  <!---->
</main>
`;
